src/NServiceBus.Core/Sagas/InvokeSagaNotFoundBehavior.cs

```diff
@@ -22,11 +24,35 @@ class InvokeSagaNotFoundBehavior : Behavior<IIncomingLogicalMessageContext>
 
             logger.InfoFormat("Could not find a started saga for '{0}' message type. Going to invoke SagaNotFoundHandlers.", context.Message.MessageType.FullName);
 
-            foreach (var handler in context.Builder.BuildAll<IHandleSagaNotFound>())
+            var nullTaskExceptions = new List<InvalidOperationException>();
```

Comment:
I considered both ways. This way they know all of the handlers that are incorrect at once, but there is a good argument to be made for simpler.